### ISSUE TITLE
Send SIGHUP to node with grps enabled process crash

### ISSUE THREAD

**Issue by dualsystems** [tendermint/tendermint#7306]
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source): 0.34.14

**ABCI app** (name for built-in, URL for self-written if it's publicly available): none

**Environment**:
- **OS** (e.g. from /etc/os-release): debian 11 

**How to reproduce it** (as minimally and precisely as possible): kill -SIGTERM  process_id

**Logs (paste a small part showing an error (< 10 lines) or link a pastebin, gist, etc. containing more of the log file)**:
```
10:35PM INF WSJSONRPC method=subscribe module=rpc-server protocol=websocket remote={"IP":"192.168.90.31","Port":59376,"Zone":""}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa10d80]

goroutine 3913950 [running]:
github.com/tendermint/tendermint/libs/pubsub.(*Subscription).Out(0xc00171d558)
        /home/com/go/pkg/mod/github.com/tendermint/tendermint@v0.34.14/libs/pubsub/subscription.go:43
github.com/tendermint/tendermint/rpc/core.Subscribe.func1()
        /home/com/go/pkg/mod/github.com/tendermint/tendermint@v0.34.14/rpc/core/events.go:50 +0x9e
created by github.com/tendermint/tendermint/rpc/core.Subscribe
        /home/com/go/pkg/mod/github.com/tendermint/tendermint@v0.34.14/rpc/core/events.go:47 +0x485
```

**node command runtime flags**: node start

**Comment by creachadair** [comment]
Thanks for the report. I think what is likely happening here is that the signal arrived at a "bad time", initiating shutdown before the client subscription arrived. I suspect that during shutdown, it stops the pubsub service, causing [this read](https://github.com/tendermint/tendermint/blob/v0.34.14/libs/pubsub/pubsub.go#L196) to succeed,  leading the [RPC subscriber](https://github.com/tendermint/tendermint/blob/v0.34.14/rpc/core/events.go#L40) to receive a `nil` subscription without error prior to entering the loop.

If that's correct, this is mostly a matter of unlucky timing during shutdown, but it should also be easy to patch up.

**Comment by creachadair** [comment]
This instance is fixed by #7309, which will land in the next point release of v0.34.
